### PR TITLE
Implement chunk_metadata

### PR DESCRIPTION
**Pull request by andrewctam**

### CODE REVIEW

**Comment by andrewctam** [comment]
**Sample output with lower chunk size:**

(base) andrew@A-2 peernode-market % cargo run
   Compiling peernode v0.1.0 (/Users/andrew/GitHub/orcanet-rust/peernode-market)
    Finished dev [unoptimized + debuginfo] target(s) in 1.91s
     Running `target/debug/peer-node`
Orcanet Peernode CLI: Type 'help' for a list of commands
> producer add files/giraffe.jpg 10
hash: 4e35e02ad76a88f028a222af36bcde08a3deff760b05f19a9531e672f0e4f1f6, size: 16384
hash: 73572a430c3b16a7f22aa235e346eb2eeba94f83dd001b3b3220945803dda798, size: 16384
hash: 3533c9c956cff9ff8c9b3f0ecda338c7ebe9cb350159db075a7e97cf3efa94e6, size: 16384
hash: e166900ab4efdb0513517b1f748482673cef2b601c175c08ce2358978b23e287, size: 16384
hash: e0ec29a0acbe8675771224804af5c935d119e7ac81a0c73be241d79fa092fca0, size: 16384
hash: 5662a9c0d4466f3737c94b7f0107331300dbf5d6bd3fac497526f1d7ae16fe41, size: 16384
hash: ea09c76664d9866e3d1db961e845495db62998e7c1b9e37e26fe480eef9e369f, size: 16384
hash: 6e6f4718d0683911f1b5e63a749c0e418253412f15ac50bacbd50df31a416302, size: 16384
hash: 2d909ab2e11174203a1ff07d1ece1a95e5a1cd194971ea5555478129188fc824, size: 16384
hash: 8aeeb1e320ad7b572e6897c5a00a2cd164456a49fb14b5596f32c4f21d1c352d, size: 16384
hash: bd761c7157047889d14a8a956b57a0d038f967e97711f07662b648fe9888b673, size: 16384
hash: 080afd159f56aa4a7820fd2443edbc3bfa6d8cd6f9692613c05c54eefc2eddf4, size: 16384
hash: 4d2ace5b1f6f8c11093940d34467bcca842ea6c3c74a98821a2d8e5f161ca9a6, size: 16384
hash: 1fd46c424b52b992e78f871148ae3a2172e4590902efa18409df4db9ca4b21d3, size: 16384
hash: 8eaa8f9559307ffa36ae18d50049614f0508184391f4ed2aa0b1704d3e2a8b4f, size: 16384
hash: 6722e1fa2e47ed3508a7ccf5161cef57c8da2ad4824838ea9fe5670ceec51da3, size: 16384
hash: e94f681fb0bc9afbeccfa5de41cce1ce5a82f5abc847bfa6fa1ae88372a4920c, size: 16384
hash: cc425ea3265e6dfa631321ca7784700543e660c672e718ce224de5fc367fb7c1, size: 16384
hash: 244d3303f30d2f4ec7030b1c27bde6a8c6a31df3fb52aac35f32dd3bddd6a323, size: 16384
hash: edab170b096b6683f0497e61a63d682cf786019176c5312aec1f5f730fd70640, size: 16384
hash: 7274dbfbaf208173f3d2d65764a07c737406c63ced2c32586e26dd2bc9bb880d, size: 16384
hash: 19ed8ecb560f93f08a861ca156d9d7d87689b5648e18c25b316f5c1e700f4fa9, size: 16384
hash: adc0e3512760bc0cab58eee03d341dec85a304af233d91f3b08411080eefd6b0, size: 16384
hash: d5fb78e3f6e2dccb5beba16ccfb229ec3478aa5c29170f113b530c09cbfed70b, size: 16384
hash: 1f9e3c11ae4a523c983b222e273d6010cb7f28c8736187560bd1d5c784f36bcd, size: 16384
hash: 3710e5e7d11f79b9b7d7ac4c1cdb28cda15d444e2d6187f6a7627b0d44cc07b6, size: 16384
hash: 5637c02514c441a58575857d155b952a4de21239488f9496ed12618cdcf0ef6b, size: 16384
hash: e8ccf6d77fcaa545b3a2031d02a5ea2487cf9b28704e990a0f9ffb0993fdf157, size: 16384
hash: ee8183bc56fbc91ed36cc4c43f14d51befd2c09584fd59d32d787753f28e6f12, size: 16384
hash: 0cf731e0a61f0d19eb65a9e3941f48ab1fcb59c809800777d166147331e55401, size: 16384
hash: 971bed9174227c0e16d13dc4f2087ddc473d8dc669992b8f8bc10e48c2a009e5, size: 16384
hash: 6185053aa422898fb07c193641650da05c601fe66d366edcceef9224f28aaaf8, size: 16384
hash: 8fa7eb4f1c65a1453a46e012ae8e7c2d98e77b308e6f9558d52c62816019ae1f, size: 16384
hash: 5c905e0a23a395cb3dccdd3b1640c97038c6c4d32beedc5f6b79436b4307e27f, size: 16384
hash: fd31bb0350ae5e9c7a15744f270077afff3b280f12118bb58c2b6f5e369854e9, size: 16384
hash: 2c1861cbd846c000c3af0b6f8bd5c0dba87f98c015be1b6082924699ffd0d4f8, size: 16384
hash: 748d4e14ccc0a4b5054df535c8fc32bd7be1cd2012519a608c1cb7d772e5e9b5, size: 16384
hash: 79abc5135369c24419e1db529f5807f0459934bc8fbb49ea8f7c7a8429c3560e, size: 16384
hash: 0dd3d6c5518632cacc8002482ecdf4edcf3145a332d99a081cc33ff498047cfb, size: 16384
hash: 3bff1bb405d7b4ed951735e12a53663d7f87b8431040ead9a37870bae4e4a998, size: 16384
hash: 116885238ced427b449335f5a2d1ce256ffeabde187f07484c522ee581a0c286, size: 16384
hash: d745655b958a744ce5a94f0b333f4d509dcef8952638653abb81dced60edceaa, size: 16384
hash: 2df2e8f5451c2bdcf558e821d4436eb7d683fe02990cb0a4a737631f904748e1, size: 16384
hash: e39824a2b178cc99f1b4a3e9eea543186ba2969e3be0a8e5018956646c02e89d, size: 16384
hash: 55a324152efd49a17bf1eb0d29f594b96360623ddd46cb50cfe533d645b5bd33, size: 16384
hash: c50b80d3aa28083991eeee9039e5eb162edf06f24987917a5011839b02b176f0, size: 16384
hash: 144334da1b92b0d09740a09b9304abf74d717d1601685a70a376c67c280fcd8f, size: 16384
hash: 29d78dbe1ffcfcc4b3e3060c4f6595114f41ca01b4d6ca3dee8b0ffce68be87a, size: 16384
hash: 3c9c964701f8fc3a56c89efa50795123820557a82ff48a45ba33dbaa487e5842, size: 16384
hash: 2c9aebf81c3136c8e88b34c99f6eea2ba4e7e5a7daf473ee498b7b66e8fab161, size: 16384
hash: 2b8ea8bfba2086062117f984241fb690351919ffc3dbafa27aa5aab7abd2aafa, size: 16384
hash: 3269aa8be6136a81ee7a830ed97bcf10ce8d9a08ce5954357673585fc89a8cf5, size: 16384
hash: a482f05c033f6e07764a4c8e276b36516f6d83d99e9ce8407d47113dae9571f6, size: 16384
hash: 95a43e24cf628d0993b5f454db5e16f33eab8eb4d12677255eb0e3d48be95457, size: 16384
hash: 3a70b46fc777983a87461c1b6ac9d47aca4ab2d62f1d9951653360bf363e2e41, size: 16384
hash: 86ef90b3383214c78f86acc072845a027f3df726bf1f3e991c585168f5ea57bf, size: 16384
hash: 53715c145076a964053bcd380cbb4c342b3dcaebcaa9b2e6e05b6fedba3efb6b, size: 16384
hash: 18178e5fede107c603964d6df399472a04d3a892137ca1ac41af2ed84214b9a6, size: 16384
hash: 266289ac7a0afd5d3f266d074872277db05130158260121f0130322f5031f5d1, size: 16384
hash: 470c5d9f2ca60febf7776f084de7d5cf6f0ccd23966182d8f1fe3b614d7417e3, size: 16384
hash: 63825641bd5a9389450a487df03f3f8a48b3b77ea62af384c23f9adbf7a25ba4, size: 16384
hash: 9502918d1b4aaa08d732338509f5dfec7fdc38c33d5941ec47c545d507ec04f5, size: 16384
hash: 7e80c6f89c5da477aa894949f22aeaca189db001db923e09762d455f651f2f72, size: 16384
hash: 61acbf450ad1fab99383a22146e116f52ca44749d2b39650f2d62d77948cda29, size: 16384
hash: 5c86adf8bbc4c6c50080b3e94aa23292d18022b3a3e53da6ff831c4df6cd19a1, size: 16384
hash: 4c0b6ed8040289ab4b74f4807fddf48d19eba323f2869119836c9a1daa64c6b5, size: 16384
hash: d036986274ec2352b0663d015cbebe63e631273028b3b1824c1e914b4365861e, size: 16384
hash: 90517685603edc6242e3e4c959b71f68001032aa9b8539e54a51d891da745148, size: 16384
hash: 188762e8e1b8e6f54e524210064dc3ffb21c0acbc4a8b30f4aca576eee6e8539, size: 16384
hash: f8e6072856ffdb092c83b4c61228d084e426e096a9afbb0664a58f4f363c542d, size: 16384
hash: 7c3dd809245da34c140166e1b1acf009976e8a741c3d5392cd101961a911ed8b, size: 16384
hash: 8c679fbaa3a384196adf87937b58c893304370c5b35f77f83cdb897b030e8fc2, size: 4194
File files/giraffe.jpg has been registered at price 10
>

**Review by daminals — APPROVED** [review]
LGTM
@snitski take a look and see if this is good for your hash checksum pr

**Review by xiej2520 — APPROVED** [review]
LGTM

**Comment by andrewctam** [comment]
@snitski I'm going to merge this PR now since this isn't going to the main branch yet. 

If you have comments, you can leave them here and we can fix those in a future PR.

File: peernode-market/src/producer/files.rs [hunk 1]
```diff
@@ -28,6 +29,26 @@ pub fn hash_file(file: &mut File) -> Result<String> {
...
+    
+    let mut sha256 = Sha256::new();
+    let mut buffer = vec![0; FileAccessType::CHUNK_SIZE as usize];
+    
+    loop {
+        let bytes_read = file.read(&mut buffer)?;
+        if bytes_read == 0 {
+            break;
+        }
+        sha256.update(&buffer[..bytes_read]);
```
Comment (snitski) [inline peernode-market/src/producer/files.rs:44]: Reminder for me: 
Need to look into and clarify this in the Google Doc. Pretty sure using `update` here makes the hash of chunk `i` depend on the hash of `i-1`. If another team hashes each chunk with a new hasher we could get conflicting file keys.

**Pull request merged**